### PR TITLE
I280 numbering references

### PR DESCRIPTION
**Pull request by kerimoyle**
Addresses #280 and #288 

Numbered references throughout the text so that people who are reading on paper can follow them.  

Global numbering is not supported (ie, can't use [1] etc) as that's reserved for local footnotes.  I've used [Ref_1] ... but open to other ideas ...

Rendered version is here: https://cellml-specification-dev.readthedocs.io/en/i280_numbering_references/reference/formal_only/specA2.html#cellml-and-xml

### CODE REVIEW

**Comment by MichaelClerx** [comment]
```
src/reference/sectionA_definitions.inc:8: WARNING: Inline interpreted text or phrase reference start-string without end-string.
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/michael/dev/cellml/specification/src/reference/formal_sectionD.rst:5: WARNING: Citation [Ref_2] is not referenced.
/home/michael/dev/cellml/specification/src/reference/formal_sectionD.rst:9: WARNING: Citation [Ref_4] is not referenced.
```

**Comment by kerimoyle** [comment]
Thanks!

**Comment by MichaelClerx** [comment]
> Thanks!

I've got a fix for some of those if you like :D

**Comment by hsorby** [comment]
Would named references be better here than using numbers?

**Comment by kerimoyle** [comment]
Yeah, I wondered about that.  They were originally named (using the same terms as their references in the conf.py file).  The trouble is that the names are what are displayed, and they were all quite long with no obvious way to shorten them?

**Comment by MichaelClerx** [comment]
Or we could just manually add a "``(see :numref:`{number}. {name}<chapter 4>`)``" everywhere?

**Comment by hsorby** [comment]
> The trouble is that the names are what are displayed,

I thought so I don't want to see that particularly.

**Comment by MichaelClerx** [comment]
> Or we could just manually add a "`` (see :numref:`{number}. {name}<chapter 4>`) ``" everywhere?

@kerimoyle would this be ok? I don't find the "[Ref 7]" links everywhere very pretty or intuitive

**Comment by kerimoyle** [comment]
What would your {name} be? The title of the reference? Won't that be really long?

**Comment by MichaelClerx** [comment]
Chapter 4. Just "see 4. References"

**Comment by kerimoyle** [comment]
Hmm.  Let me try some stuff.

**Comment by MichaelClerx** [comment]
I'm beginning to think this issue 

- users who print the spec don't get a numbered link to external hyperlinks they'll never manually type into a browser anyway

isn't a big enough issue to warrant the kind of changes it seems to imply?

**Comment by MichaelClerx** [comment]
If it interferes with readability for PDF or HTML users I don't really want to do it

**Comment by kerimoyle** [comment]
Yeah, I get it, I just have an idea I want to try.  If that doesn't work out will ditch it.

**Comment by MichaelClerx** [comment]
Thanks!

**Comment by MichaelClerx** [comment]
I don't mind how that looks!

**Comment by kerimoyle** [comment]
Gosh, high praise indeed :D

**Comment by MichaelClerx** [comment]
:D 

It gets worse! Should we renumber so the refs are in order?

**Comment by kerimoyle** [comment]
They're currently alphabetical by title?

**Comment by MichaelClerx** [comment]
ok!

**Comment by kerimoyle** [comment]
See, now you're getting delirious ... :)

**Comment by agarny** [comment]
It started a bit earlier. He's sleep deprived.

File: src/reference/sectionA_definitions.inc [hunk 1]
```diff
@@ -50,7 +50,7 @@ Any information item in the :ref:`CellML namespace<specA_cellml_namespace>`.
 .. _specA_basic_latin_alphabetical_character:
 
 **Basic Latin alphabetical character:**
-A Unicode character in the range :unicode:`0041` to :unicode:`005A` or in the range :unicode:`0061` to :unicode:`007A`.
+A Unicode [:ref:`5<ref_unicode>`] character in the range :unicode:`0041` to :unicode:`005A` or in the range :unicode:`0061` to :unicode:`007A`.
```
Comment (hsorby) [inline src/reference/sectionA_definitions.inc:53]: I think the `Unicode` word here should be a link.

**Comment by MichaelClerx** [comment]
I'd also prefer order of appearance, but I'll echo Alan's "meh"

> meh!

File: src/reference/sectionA_definitions.inc (hunk 1 above)
Comment (kerimoyle) [reply src/reference/sectionA_definitions.inc:53]: Good idea.

Comment (kerimoyle) [reply src/reference/sectionA_definitions.inc]: Done in 91f99aa

**Comment by MichaelClerx** [comment]
Click the button! Click the button! Quick!

**Comment by kerimoyle** [comment]
Just let me check that the mehs have been fixed ...

**Comment by MichaelClerx** [comment]
> Just let me check that the mehs have been fixed ...

Beautiful! I rescind my echoed meh!

**Pull request merged**